src/mongo_client.ts

```diff
@@ -554,3 +572,92 @@ export class MongoClient extends EventEmitter implements OperationParent {
+ * @internal
+ */
+export interface MongoOptions
+  extends Required<BSONSerializeOptions>,
+    Omit<ConnectionOptions, 'port'>,
+    Omit<TLSConnectionOptions, 'port'>,
+    Required<
+      Pick<
+        MongoClientOptions,
+        | 'autoEncryption'
+        | 'compression'
+        | 'compressors'
+        | 'connectTimeoutMS'
+        | 'dbName'
+        | 'directConnection'
+        | 'domainsEnabled'
+        | 'driverInfo'
+        | 'forceServerObjectId'
+        | 'gssapiServiceName'
+        | 'ha'
+        | 'haInterval'
+        | 'heartbeatFrequencyMS'
+        | 'keepAlive'
+        | 'keepAliveInitialDelay'
+        | 'localThresholdMS'
+        | 'logger'
+        | 'maxIdleTimeMS'
+        | 'maxPoolSize'
+        | 'minPoolSize'
+        | 'monitorCommands'
+        | 'noDelay'
+        | 'numberOfRetries'
+        | 'pkFactory'
+        | 'promiseLibrary'
+        | 'raw'
+        | 'reconnectInterval'
+        | 'reconnectTries'
+        | 'replicaSet'
+        | 'retryReads'
+        | 'retryWrites'
+        | 'serverSelectionTimeoutMS'
+        | 'serverSelectionTryOnce'
+        | 'socketTimeoutMS'
+        | 'tlsAllowInvalidCertificates'
+        | 'tlsAllowInvalidHostnames'
+        | 'tlsInsecure'
+        | 'waitQueueMultiple'
+        | 'waitQueueTimeoutMS'
+        | 'zlibCompressionLevel'
+      >
+    > {
+  hosts: { host: string; port: number }[];
+  srv: boolean;
+  credentials: MongoCredentials;
+  readPreference: ReadPreference;
+  readConcern: ReadConcern;
+  writeConcern: WriteConcern;
+
+  /**
+   * # NOTE ABOUT TLS Options
+   *
+   * If set TLS enabled, equivalent to setting the ssl option.
+   *
+   * ### Additional options:
+   *
+   * |    nodejs option     | MongoDB equivalent                                       | type                                   |
+   * |:---------------------|--------------------------------------------------------- |:---------------------------------------|
+   * | `ca`                 | `sslCA`, `tlsCAFile`                                     | `string \| Buffer \| Buffer[]`         |
+   * | `crl`                | `sslCRL`                                                 | `string \| Buffer \| Buffer[]`         |
+   * | `cert`               | `sslCert`, `tlsCertificateFile`, `tlsCertificateKeyFile` | `string \| Buffer \| Buffer[]`         |
+   * | `key`                | `sslKey`, `tlsCertificateKeyFile`                        | `string \| Buffer \| KeyObject[]`      |
+   * | `passphrase`         | `sslPass`, `tlsCertificateKeyFilePassword`               | `string`                               |
+   * | `rejectUnauthorized` | `sslValidate`                                            | `boolean`                              |
+   *
+   */
+  tls: boolean;
+
+  /**
+   * Turn these options into a reusable options dictionary
+   */
+  toJSON(): Record<string, any>;
+  /**
+   * Turn these options into a reusable connection URI
+   */
+  toURI(): string;
```

Comment:
```suggestion
  toConnectionString(): string;
```
just a thought? I'm the one that introduced calling it a URI, but I think the consensus across the drivers and the company is calling it a connection string

Reply:
[Connection String URI Format](https://docs.mongodb.com/manual/reference/connection-string/)
It's one of those both ways is right, I'll leave it as is, and we can consider renaming it when we get to the ticket that implements it since there might be some questions about how we want to expose it. These options are supposed to be internal, kind of, so is it something we'd want people to use: `client.options.toURI()` or would we surface something on the client itself.